src/rpc/rawtransaction.cpp

```diff
@@ -1160,13 +1160,35 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
     CValidationState state;
     bool missing_inputs;
     bool test_accept_res;
+    CAmount fee_out;
     {
         LOCK(cs_main);
         test_accept_res = AcceptToMemoryPool(mempool, state, std::move(tx), &missing_inputs,
-            nullptr /* plTxnReplaced */, false /* bypass_limits */, max_raw_tx_fee, /* test_accept */ true);
+            nullptr /* plTxnReplaced */, false /* bypass_limits */, max_raw_tx_fee, /* test_accept */ true, /* fee_out */ &fee_out);
     }
     result_0.pushKV("allowed", test_accept_res);
     if (!test_accept_res) {
+        auto rc = state.GetRejectCode();
+        if (rc == REJECT_HIGHFEE || rc == REJECT_INSUFFICIENTFEE) {
+            UniValue fees(UniValue::VOBJ);
+            CFeeRate fr_actual(fee_out, (weight+3)/4);
+            CFeeRate fr_max(max_raw_tx_fee, (weight+3)/4);
+            fees.pushKV("fee-txsize", (uint64_t)(weight+3)/4);
+            fees.pushKV("fee-abs-actual", ValueFromAmount(fee_out));
+            fees.pushKV("fee-abs-max", ValueFromAmount(max_raw_tx_fee));
+            fees.pushKV("fee-rate-actual", ValueFromAmount(fr_actual.GetFeePerK()));
+            fees.pushKV("fee-rate-max", ValueFromAmount(fr_max.GetFeePerK()));
+            if (rc == REJECT_HIGHFEE) {
+                CFeeRate fr_excess(fee_out - max_raw_tx_fee, (weight+3)/4);
+                fees.pushKV("fee-abs-excess", ValueFromAmount(fee_out - max_raw_tx_fee));
+                fees.pushKV("fee-rate-excess", ValueFromAmount(fr_excess.GetFeePerK()));
+            } else {
+                CFeeRate fr_shortage(max_raw_tx_fee - fee_out, (weight+3)/4);
+                fees.pushKV("fee-abs-shortage", ValueFromAmount(max_raw_tx_fee - fee_out));
+                fees.pushKV("fee-rate-shortage", ValueFromAmount(fr_shortage.GetFeePerK()));
+            }
+            result_0.pushKV("fees", fees);
```

Comment:
in commit 349860df8695ff0ad657a3d6d1502edb46571174

This is undocumented, but I am also not sure if it is a desired feature. Might want to split out as a separate pull request after the first part is merged?

Reply:
I agree that this is unnecessary in this PR. Happy to review in a follow-up PR.